contents/tutorials/evaluation-runtimes-and-tags.md

```diff
@@ -0,0 +1,335 @@
+
+## Setting up both features
+
+### Step 1: Configure the flag in PostHog
+
+When creating or editing a feature flag:
+
+1. **Set the evaluation runtime**:
+   - Find the "Evaluation runtime" selector
+   - Choose `server`, `client`, or `all` (the default is `all`)
+
+2. **Add evaluation tags**:
+   - Click on "Add tags" and create your tag(s)
+   - Click the bolt icon ⚡ to configure evaluation constraints
+   - Select which tags should act as evaluation constraints
+
+### Step 2: Configure your SDKs
+
+Update your SDK initialization to include evaluation environments. Here's how to set it up for different scenarios:
+
+**Production API server (Node.js):**
+
+```javascript
+const posthog = new PostHog('YOUR_API_KEY', {
+    host: 'https://app.posthog.com',
+    evaluation_environments: ['production', 'api', 'backend']
+})
+```
+
+**Production web app (React):**
+
+```jsx
+<PostHogProvider 
+    apiKey='YOUR_API_KEY'
+    options={{
+        api_host: 'https://app.posthog.com',
+        evaluation_environments: ['production', 'web', 'frontend']
+    }}
+>
+    <App />
+</PostHogProvider>
+```
+
+**Staging microservice (Python):**
+
+```python
+posthog = Posthog(
+    'YOUR_API_KEY',
+    host='https://app.posthog.com',
+    evaluation_environments=['staging', 'payments-service', 'backend']
+)
+```
+
+## Common use cases
+
+### 1. Sensitive server-only features with environment isolation
+
+**Scenario**: Database migration flags that should never leak to clients and only run in specific environments.
+
+**Configuration**:
+
+- Runtime: `server`
+- Evaluation tags: `["production", "migration-runner"]`
+
+**Result**: Only server-side SDKs in production with the "migration-runner" tag can evaluate this flag.
+
+```javascript
+// Migration service
+const posthog = new PostHog('KEY', {
+    evaluation_environments: ['production', 'migration-runner']
+})
+
+// This service can evaluate the migration flag
+const shouldRunMigration = await posthog.isFeatureEnabled('db-migration-v2')
+```
+
+### 2. Platform-specific client features
+
+**Scenario**: Mobile-only features that should work differently in staging vs production.
+
+**Configuration**:
+
+- Runtime: `client`
+- Evaluation tags: `["mobile", "ios"]`
+
+**Production iOS app:**
+
+```swift
+// Production iOS app gets the flag
+posthog.setup(apiKey: "KEY", host: "https://app.posthog.com", 
+              evaluationEnvironments: ["production", "mobile", "ios"])
+```
+
+**Staging iOS app:**
+
+```swift
+// Staging iOS app doesn't get this flag (missing "production" tag)
+posthog.setup(apiKey: "KEY", host: "https://app.posthog.com",
+              evaluationEnvironments: ["staging", "mobile", "ios"])
+```
+
+### 3. Cross-platform features with environment control
+
+**Scenario**: New checkout flow that needs testing across web and mobile, but only in staging.
+
+**Configuration**:
+
+- Runtime: `all`
+- Evaluation tags: `["staging", "checkout"]`
```

Comment:
> only in staging

Given the presence of `checkout`, this would also match a flag tagged as `production`, `checkout`.